File: FreelanceMarketplace/FreelanceMarketplace/GraphQL/Types/NotificationType.cs
using FreelanceMarketplace.Models;
using GraphQL.Types;

namespace FreelanceMarketplace.GraphQL.Types
{
    public class NotificationType : ObjectGraphType<Notification>
    {
        public NotificationType()
        {
            Field(x => x.Id);
            Field(x => x.SenderId);
            Field(x => x.ReceiverId);
            Field(x => x.Message);
            Field(x => x.CreatedAt);
            Field(x => x.IsRead);

            Field<UserType>("sender");
            Field<UserType>("receiver");
        }
    }
}

File: FE/src/types/NotificationType.ts
import { User } from "./UserType";

export interface NotificationItem {
  id: number;
  sender: string;
  recipient: string;
  message: string;
  createdAt: string;
  isRead: boolean;
}

export interface NotificationType {
  id: number;
  message: string;
  isRead: boolean;
  createdAt: string;
  senderId: number;
  receiverId: number;
  sender: User;
  receiver: User;
}